rpc/block_test.go

```diff
@@ -702,3 +702,119 @@ func TestRpcBlockAdaptation(t *testing.T) {
 		require.Equal(t, &felt.Zero, blockWithTxHashes.BlockHeader.SequencerAddress)
 	})
 }
+
+func TestJunoGetBlockWithTxsAndReceipts(t *testing.T) {
+	blockId := map[string]rpc.BlockID{
+		"latest":  {Latest: true},
+		"pending": {Pending: true},
+		"hash":    {Hash: new(felt.Felt).SetUint64(1)},
+		"number":  {Number: 1},
+	}
+
+	for desc, id := range blockId {
+		t.Run(desc, func(t *testing.T) {
+			log := utils.NewNopZapLogger()
+			n := utils.Ptr(utils.Mainnet)
+			chain := blockchain.New(pebble.NewMemTest(t), n)
+			handler := rpc.New(chain, nil, nil, "", n, log)
+
+			block, blockReceipt, rpcErr := handler.JunoGetBlockWithTxsAndReceipts(id)
+			assert.Nil(t, block)
+			assert.Nil(t, blockReceipt)
+			assert.Equal(t, rpc.ErrBlockNotFound, rpcErr)
```

Comment:
We should have an expected response, to make sure that the handler returns what we expect. Currently we just test that we get _some_ response, not whether it is the correct response

Reply:
For exmaple, see above
assert.Equal(t, &rpc.BlockWithReceipts{
			Status: rpc.BlockPending,
			BlockHeader: rpc.BlockHeader{
				Hash:             header.Hash,
				ParentHash:       header.ParentHash,
				Number:           header.Number,
				NewRoot:          header.NewRoot,
				Timestamp:        header.Timestamp,
				SequencerAddress: header.SequencerAddress,
				L1GasPrice:       header.L1GasPrice,
				StarknetVersion:  header.StarknetVersion,
			},
			Transactions: txsWithReceipt,
		}, resp)